var/spack/repos/builtin/packages/trexio/package.py

```diff
@@ -55,10 +55,10 @@ class AutotoolsBuilder(autotools.AutotoolsBuilder):
     def configure_args(self):
         config_args = []
         if "+hdf5" in self.spec:
-            if self.spec("@:2.3.0"):
+            if self.spec.satisfies("@:2.3.0"):
                 config_args.append("LIBS=-lhdf5_hl")
             else:
-                config.args.append("LIBS=-lhdf5")
+                config_args.append("LIBS=-lhdf5")
```

Comment:
Technically, Autotools takes care of that and adds the neccessary `-l` and `-L` flags when HDF5 is detected (which was not always the case for the `hdf5_hl` hence the first `if`. 

I have nothing against leaving this line if it works.

Thank you for fixing, LGTM!

Reply:
I've no objection removing it either. @RMeli your call.

Reply:
Thanks for the review @q-posev! I didn't change the logic of what was there, I only fixed the issues. With your insight, I'm happy to remove the `else` branch.